theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -996,6 +996,8 @@ def get_params(self, node):
     def perform(self, node, axis_and_tensors, out_, ctx):
         out, = out_
         axis = int(axis_and_tensors[0])
+        if axis < 0:
+            axis += axis_and_tensors[1].ndim
```

Comment:
I don't see the check you are talking about, and anyway, it fails because of an OverflowError even before that, because `axis` is declared as an `unsigned int`.
The existing tests check that the error is not covered up.